include/boost/iostreams/filter/zlib.hpp

```diff
@@ -347,7 +347,7 @@ bool zlib_compressor_impl<Alloc>::filter
     int result = xdeflate(flush ? zlib::finish : zlib::no_flush);
     after(src_begin, dest_begin, true);
     zlib_error::check BOOST_PREVENT_MACRO_SUBSTITUTION(result);
-    return result != zlib::stream_end; 
+    return result == zlib::okay; 
```

Comment:
According to the zlib docs, if zlib::finish is passed  in the call to deflate then zlib::stream_end is returned when successful, otherwise zlib::okay is returned when successful. I do not think your change reflects that reality.